cluster_utils/settings.py

```diff
@@ -302,6 +309,68 @@ def report_exit_at_server():
     send_message(MessageTypes.JOB_CONCLUDED, message=(submission_state.job_id,))
 
 
+def init_main_script_argument_parser(description: str) -> argparse.ArgumentParser:
```

Comment:
Actually thinking about it.  At least the the other functions below are directly related to reading settings (and also use quite a few functions from `settings.py`, so I think it would make sense to keep them where they are and really rather move the user API functions somewhere else.

Reply:
Makes sense, let's move out the user functions to a dedicated module (and import them in `__init__`). 

I would not make an subpackage for internals though, as actually most of the stuff does not concern the client. So instead I would rather make a dedicated `client` subpackage.

Related:

- https://github.com/martius-lab/cluster_utils/issues/45
- https://github.com/martius-lab/cluster_utils/issues/83

Maybe we can also already keep those in mind for making changes now.

Reply:
Is it okay if I do this in a new PR?